frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetailHooks.tsx

```diff
@@ -59,3 +62,92 @@ export function useFetchVapb() {
 
   return { vapbItems: data?.searchResult?.[0]?.items, loading, err: error?.message }
 }
+
+export function useFetchKyvernoRelated() {
+  const urlParams = useParams()
+  const name = urlParams.templateName ?? '-'
+  const kind = urlParams.kind ?? '-'
+  const apiGroup = urlParams.apiGroup ?? ''
+  const { clusterName, template, templateLoading } = useTemplateDetailsContext()
+  const [getKyverno, { loading, error, data }] = useSearchResultRelatedItemsLazyQuery({
+    client: process.env.NODE_ENV === 'test' ? undefined : searchClient,
+  })
+  const [filtered, setFiltered] = useState<any[]>()
+  useEffect(() => {
+    if (
+      apiGroup === 'kyverno.io' &&
+      clusterName &&
+      name &&
+      template &&
+      !templateLoading &&
+      // These conditions reduce renders and hitting fireManagedClusterView
+      !loading &&
+      !error &&
+      !data
+    ) {
+      getKyverno({
+        client: process.env.NODE_ENV === 'test' ? undefined : searchClient,
+        variables: {
+          input: [
+            {
+              filters: [
+                {
+                  property: 'apigroup',
+                  values: ['kyverno.io'],
+                },
+                {
+                  property: 'kind',
+                  values: [kind],
+                },
+                {
+                  property: 'name',
+                  values: [name],
+                },
+                {
+                  property: 'cluster',
+                  values: [clusterName],
+                },
+              ],
+              limit: 1000000,
+            },
+          ],
+        },
+      })
+    }
+  }, [apiGroup, clusterName, name, template, templateLoading, data, loading, error, kind, getKyverno])
+
+  useEffect(() => {
+    if (data) {
+      // Attach a PolicyReport to a related resource.
+      const reportMap = data.searchResult?.[0]?.related
+        ?.filter((r) => ['PolicyReport', 'ClusterPolicyReport'].includes(r?.kind ?? ''))
+        .map((r) => r?.items)
+        .flat()
+        .reduce((accumulator, currentValue) => ({ ...accumulator, [currentValue.name]: currentValue }), {})
+
+      setFiltered(
+        data?.searchResult?.[0]?.related
+          ?.map((related) => related?.items)
+          .flat()
+          .filter((item) => !['PolicyReport', 'ClusterPolicyReport', 'Cluster'].includes(item.kind))
```

Comment:
```suggestion
          .filter((item) => item._uid && !['PolicyReport', 'ClusterPolicyReport', 'Cluster'].includes(item.kind))
```

Reply:
_uid is must have why do we need?

Reply:
I saw below that you had `item._uid?.split('/') || ''` so I thought it was some kind of linting issue you were working around.

Reply:
Ah because Typescript. I don't need "item._uid?" but typescript error so